lib/workers/repository/process/write.ts

```diff
@@ -40,7 +47,30 @@ export async function writeUpdates(
     }
     addMeta(meta);
     const branchExisted = branchExists(branch.branchName);
-    const res = await processBranch(branch);
+    let branchCache = {} as BranchCache;
+    if (branchExisted) {
+      branchCache =
+        cachedBranches?.find((br) => br.branchName === branch.branchName) ??
+        ({} as BranchCache);
+
+      if (
+        config.repositoryCache === 'enabled' &&
+        Object.keys(branchCache).length === 0
+      ) {
+        logger.debug(`No branch cache found for ${branch.branchName}`);
+      }
+
+      const branchManagersFingerprint = hasha(
+        branch.upgrades
+          .map((upgrade) => hashMap.get(upgrade.manager) ?? upgrade.manager)
+          .filter(is.string)
+      );
+      branch.branchFingerprint = hasha([
+        JSON.stringify(config),
+        branchManagersFingerprint,
+      ]);
```

Comment:
But did we just introduce a problem because of my changes?

On the run where a branch is created then we *do* want the fingerprint written to cache. Is that the case?

Reply:
No, the first time we create a branch we dont store the fingerprint in cache after your changes.